pkg/liqo-controller-manager/namespaceMap-controller/namespaceMap_controller.go

```diff
@@ -45,42 +46,46 @@ const (
 )
 
 // Reconcile adds/removes NamespaceMap finalizer, and checks differences
-// between DesiredMapping and CurrentMapping.
+// between DesiredMapping and CurrentMapping in order to create/delete remote Namespaces if it is necessary.
 func (r *NamespaceMapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	namespaceMap := &mapsv1alpha1.NamespaceMap{}
-	if err := r.Get(context.TODO(), req.NamespacedName, namespaceMap); err != nil {
+	if err := r.Get(ctx, req.NamespacedName, namespaceMap); err != nil {
 		klog.Errorf("%s --> Unable to get NamespaceMap '%s'", err, req.Name)
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
 	if !ctrlutils.ContainsFinalizer(namespaceMap, liqoconst.NamespaceMapControllerFinalizer) {
 		ctrlutils.AddFinalizer(namespaceMap, liqoconst.NamespaceMapControllerFinalizer)
-		if err := r.Patch(context.TODO(), namespaceMap, client.Merge); err != nil {
+		if err := r.Patch(ctx, namespaceMap, client.Merge); err != nil {
 			klog.Errorf("%s --> Unable to add finalizer to the NamespaceMap '%s'", err, namespaceMap.GetName())
 			return ctrl.Result{}, err
 		}
 		klog.Infof("Finalizer correctly added on NamespaceMap '%s'", namespaceMap.GetName())
 	}
 
-	if err := r.manageRemoteNamespaces(namespaceMap); err != nil {
+	requeue, err := r.manageRemoteNamespaces(namespaceMap)
+	if err != nil {
 		return ctrl.Result{}, err
 	}
 
 	if len(namespaceMap.Status.CurrentMapping) == 0 {
 		ctrlutils.RemoveFinalizer(namespaceMap, liqoconst.NamespaceMapControllerFinalizer)
-		if err := r.Update(context.TODO(), namespaceMap); err != nil {
+		if err := r.Update(ctx, namespaceMap); err != nil {
 			klog.Errorf("%s --> Unable to remove '%s' from NamespaceMap '%s'", err,
 				liqoconst.NamespaceMapControllerFinalizer, namespaceMap.GetName())
 			return ctrl.Result{}, err
 		}
 		klog.Infof("Finalizer correctly removed from NamespaceMap '%s'", namespaceMap.GetName())
 	}
 
+	if requeue {
+		return ctrl.Result{}, fmt.Errorf("unable to create some remote namespace")
+	}
 	return ctrl.Result{}, nil
 }
 
-// The Controller is triggered only when the number of entries in DesiredMapping changes, so only when
-// a namespace's request is added or removed.
+// Events not filtered:
```

Comment:
The refactor of this comment is not really easy to understand IMHO.

Reply:
I will rewrite it, thank you